huaweicloud/services/coc/resource_huaweicloud_coc_war_room.go

```diff
@@ -0,0 +1,450 @@
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"recover_member": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"recover_leader": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"incident": buildWarRoomsReqIncident(),
+			"source": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"change_num": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"occur_time": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"recover_time": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"fault_cause": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"create_time": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"first_report_time": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"recovery_notification_time": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"fault_impact": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"circular_level": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"war_room_status": buildWarRoomsReqWarRoomStatus(),
+			"processing_duration": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"restoration_duration": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildWarRoomsReqIncident() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"id": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"incident_id": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"is_change_event": {
+					Type:     schema.TypeBool,
+					Computed: true,
+				},
+				"failure_level": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"incident_url": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+}
+
+func buildWarRoomsReqWarRoomStatus() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"id": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"name_zh": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"name_en": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"type": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+}
+
+func buildWarRoomCreateOpts(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"war_room_name":         d.Get("war_room_name"),
+		"description":           utils.ValueIgnoreEmpty(d.Get("description")),
+		"region_code_list":      utils.ValueIgnoreEmpty(d.Get("region_code_list").([]interface{})),
```

Comment:
no need, check other place
```suggestion
		"region_code_list":      utils.ValueIgnoreEmpty(d.Get("region_code_list")),
```